### PR TITLE
Fix for: [Interface Skeleton] Limit the editor width to prevent some blocks to grow infinitely wide

### LINKED ISSUE #27622: Interface Skeleton: Limit the editor width to prevent some blocks to grow infinitely wide
**Describe the bug**

With #25884, some blocks were somehow able to exceed the editor boundaries and, when set to full width, gain a width of literally millions of pixels, making the editor unusable.

In #26552 we added a `max-width: 100%` to `.interface-interface-skeleton__editor` which seemed to fix the issue.

The fix was then removed in #26944 because not compatible with IE11.

So, again some blocks can become infinitely wide.
Empirically, this seems to happen with slideshow blocks (see for example the Slideshow block included in Jetpack).

**To reproduce**

Note: I'll mention Jetpack in the test steps because it's the one plugin I'm familiar with.
Users have reported that the issue affects other slideshow blocks, but they might not necessarily be openly available for testing.
Feel free to comment with details on how to test with other blocks, and I'll be happy to update these steps to include them as well.

1. Install and activate Jetpack.
2. Open the post editor and insert a Slideshow block. Add some images to it.
3. Open the browser's dev tools, search for `.interface-interface-skeleton__editor` and disable the `max-width: 100%` rule. That is provided by Jetpack exactly to fix this issue.
4. Set the Slideshow block at full width.
5. The block is now millions of pixel wide, basically breaking the editor.

**Expected behavior**

The slideshow block should have been as wide as the editor.

**Screenshots**

| Normal Width | Full Width | Full Width (Highlighted) |
| - | - | - |
| Screenshot 2020-12-09 at 18 00 24 | Screenshot 2020-12-09 at 18 00 35 | Screenshot 2020-12-09 at 18 00 55 |

**Editor version (please complete the following information):**
- WordPress version: 5.6 (tested up to 5.7-alpha-49774)
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? Both gutenberg plugin and default.
- If the Gutenberg plugin is installed, which version is it? 9.5.1

**Desktop (please complete the following information):**
 - OS: MacOS Big Sur 11.0.1
 - Browser: Chrome
 - Version: 87

**Additional context**

Given the original fix was removed because not compatible with IE11, let's make sure to test the new fix on IE11 as well!

### LINKED ISSUE #28140: Editor sidebar clipped when block content is too wide
**Describe the bug**

With wide enough block, the sidebar is clipped.

**To reproduce**

Add a Heading block, input a long space-less string. Observe the block content pushes the sidebar to the side, clipping it. This happens on screen widths between 780px and 1070px.

**Expected behavior**

The sidebar should not be clipped.

**Screenshots**

https://d.pr/i/thsAt5

**Editor version (please complete the following information):**
- WordPress version: 5.6
- Default Gutenberg editor

**Desktop (please complete the following information):**
 - OS: macOS
 - Browser: Chrome
 - Version: 87

### PR DESCRIPTION
**Pull request by ambienthack**
Fixes #27622. Another try to prevent interface from overflowing browser window when Jetpack Slideshow block is used.

## Description
```css
.interface-interface-skeleton__editor {
        ...
	flex: 0 1 100%; 
        overflow: hidden; /* <- added */
}
```
When a large non-breaking element (width>100% of its container) is placed inside of a flex element, by default flex element grows beyond 100% trying to become as large as the content, even ignoring `flex-grow:0`. _I think_ there is some kind of inifine loop created by Swiper code (used in Jetpack Slideshow) and browser flex layout logic: Swiper tries to take all available space => flex element grows to embrace the content. `overflow: hidden;`  breaks the loop.

## How has this been tested?
Tested according to steps from #27622 
* Chrome (latest), Firefox (latest), Opera (latest), IE11 on Windows 10

Also checked that the solution works well with the Full Site Editing drawer.

Latest Jetpack Slideshow doesn't work in IE11. In IE11 I tested with jetpack 8.5 (it seems to be the last working version in IE11)
Can be downloaded here https://wordpress.org/plugins/jetpack/advanced/ (bottom of the page)

## Checklist:
- [ ] My code is tested.
- [x] My code follows the WordPress code style.

### CODE REVIEW

**Comment by webprom** [comment]
better to add `max-width: 100%;` there

**Comment by ambienthack** [comment]
i don't think max-width:100% is a good solution. see details in the comment 
https://github.com/WordPress/gutenberg/pull/27646#issuecomment-743553418

**Comment by noisysocks** [comment]
Thanks for the PR!

Paging @jasmussen as he's my go-to expert on what repercussions this change might have 🙂

**Comment by talldan** [comment]
I tested and it looks like this fixes https://github.com/WordPress/gutenberg/issues/28140.

**Comment by jasmussen** [comment]
Thank you for the continued work on these really intricate but fundamentally important details @ambienthack, really appreciate it. (And thank you for the ping, Robert!)

In this case, I don't have my Jetpack testing environment up and running, so I focused my testing on whether this was causing any regressions in the block editor interface and basic content. Doesn't seem like it did. I tested all breakpoints, with and without fullscreen mode. Seems to be fine both in the post editor and site editor 🎉 

I also considered the case where a theme developer might intentionally want the entire website to scroll horizontally (a la https://wordpress.com/theme/shelf), but that should still be possible, as that touches the editor styles wrapper inside.

It might be worth rebasing this before merging, then testing again. Not just to give the tests a second chance, but because the site editor is now in an iframe, and that might affect the skeleton. I don't think it should, if anything that might on its own solve the issue, but it's something to be aware of. 

Nice work.

**Comment by ambienthack** [comment]
Thank you guys for kind words! I've rebased the PR. I'm not able to test it heavily right now. Only checked the basic case: post editor.

**Comment by jasmussen** [comment]
Thanks for the rebase. Happy to take this for another spin in the site editor tomorrow and approve, unless someone beats me to it.

**Review by jasmussen — APPROVED** [review]
Tested in post and site editors after rebase. No visual change, so it seems fine to try, especially because it's such a small change. Let's get it in, thank you for your contribution!

**Pull request merged**
